### PR TITLE
Add fml::paths::GetPath and a Linux backend.

### PR DESCRIPTION
**Pull request by chinmaygarde**
For initial review. Adding the other backends now.

### CODE REVIEW

File: fml/paths.cc [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+#endif
+
+namespace fml {
+namespace paths {
+
+std::pair<bool, std::string> GetPath(PathType type) {
+  static std::once_flag once;
+  static PathProvider* provider = nullptr;
+  std::call_once(once, []() { provider = new PlatformPathProvider(); });
```
Comment (abarth) [inline fml/paths.cc]: We don't have atomically initialized statics?

**Comment by abarth** [comment]
Why do this with virtual dispatch instead of with ifdefs?

**Comment by chinmaygarde** [comment]
> Why do this with virtual dispatch instead of with ifdefs?

Just continuing the pattern followed elsewhere. Admittedly, virtual dispatch is not needed here (at least for the Linux backend, Android may have some one time setup though I am not sure yet).

**Comment by chinmaygarde** [comment]
Added PathProvider backend for Android and Darwin. The virtual dispatch was not necessary. Should I get rid of it or just stick to the pattern?

**Comment by chinmaygarde** [comment]
Getting rid of it now.

File: fml/paths.h [hunk 2]
```diff
@@ -0,0 +1,20 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_FML_PATH_SERVICE_H_
+#define FLUTTER_FML_PATH_SERVICE_H_
+
+#include <memory>
```
Comment (abarth) [inline fml/paths.h]: memory probably isn't needed.

File: fml/paths.h [hunk 3]
```diff
@@ -0,0 +1,20 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#ifndef FLUTTER_FML_PATH_SERVICE_H_
+#define FLUTTER_FML_PATH_SERVICE_H_
```
Comment (abarth) [inline fml/paths.h]: Drop service

**Comment by abarth** [comment]
LGTM

**Pull request merged**